features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/di/TimelineItemPresenterFactories.kt

```diff
@@ -50,9 +52,18 @@ interface TimelineItemPresenterFactoriesModule {
  *
  * Request this type from Dagger to access the [TimelineItemPresenterFactory] map multibinding.
  */
+@SingleIn(RoomScope::class)
 data class TimelineItemPresenterFactories @Inject constructor(
     val factories: @JvmSuppressWildcards Map<Class<out TimelineItemEventContent>, TimelineItemPresenterFactory<*, *>>,
-)
+) {
+    /**
+     * Stores the presenter states for each event id.
+     *
+     * This is so TimelineItemPresenters can store their state in a place which is not
+     * affected by going in and out of the composition due to being inside a LazyColumn.
+     */
+    val presenterStates: MutableMap<EventId, Any> = mutableMapOf()
```

Comment:
I'd extract this to a separate things, as it sounds weird to inject `TimelineItemPresenterFactories` inside a `Presenter`.